### PR TITLE
Financial activity

### PR DESCRIPTION
**Pull request by lomamech**
Patient Financial Activity  
- Update bhLatestInvoice template
- Adding the route for get financial patient activity
- Adding in finance debtors the request for patient financial Activity
- Adding the report pdf for Patient Financial Activity

Personalisation of text on the button for bhPDFPrint component

@jniles

### CODE REVIEW

File: client/src/js/components/bhPDFPrint.js [hunk 1]
```diff
@@ -17,13 +17,14 @@ angular.module('bhima.components')
...
+    buttonText : '@',
     disableCache: '@',
     options : '<',
   },
   transclude : true,
-  template :
+  template : 
     '<a href ng-click="$ctrl.print()">' +
-      '<span ng-if="!$ctrl.$loading"><i class="fa fa-print"></i> {{ "FORM.BUTTONS.PRINT" | translate }}</span>' +
+      '<span ng-if="!$ctrl.$loading"><i class="fa fa-print"></i> {{ $ctrl.buttonText | translate }}</span>' +
```
Comment (mbayopanda) [inline client/src/js/components/bhPDFPrint.js:27]: What happened if the `buttonText` it is not set ? I think you must set a default value for `buttonText`.

``` javascript
// In the controller 
var component = this;

component.buttonText = component.buttonText || 'FORM.BUTTONS.PRINT';
```

File: server/controllers/finance/debtors/index.js [hunk 2]
```diff
@@ -266,7 +266,8 @@ function financialPatient(debtorUuid) {
       LEFT JOIN project ON invoice.project_id = project.id 
       WHERE general_ledger.entity_uuid = ? AND general_ledger.entity_type = 'D'
     ) as transaction
-    GROUP BY transaction.trans_id;`;
+    GROUP BY transaction.trans_id 
+    ORDER BY transaction.trans_date ASC;`;
```
Comment (mbayopanda) [inline server/controllers/finance/debtors/index.js:270]: 👍

**Review by mbayopanda — COMMENTED** [review]
This PR works very well 👍

File: server/controllers/finance/debtors/index.js (hunk 2 above)
Comment (mbayopanda) [reply server/controllers/finance/debtors/index.js:270]: I notice that the user interface in the patient billing zone display some data when there are not any invoices for the patient 
last invoice

Could you hide these informations when there are not invoices for the patient, and display a meaningful message such as `No invoices found for the patient`  well centered in the zone for example.

File: client/src/js/components/bhPDFPrint.js (hunk 1 above)
Comment (lomamech) [reply client/src/js/components/bhPDFPrint.js:27]: If the buttontext is not set the text will be FORM.BUTTONS.PRINT

**Comment by lomamech** [comment]
@mbayopanda 
I came to fix that

**Review by sfount — COMMENTED** [review]
LGTM. @mbayopanda 's comments have been addressed.

**Pull request merged**